framework/include/materials/TwoMaterialPropertyInterface.h

```diff
@@ -114,7 +114,10 @@ TwoMaterialPropertyInterface::getNeighborADMaterialProperty(const std::string &
   if (default_property)
     return *default_property;
   else
+  {
+    _material_property_dependencies.insert(_material_data->getPropertyId(prop_name));
```

Comment:
Any idea why no dependency was added for neighbor AD properties?!

Reply:
Because someone was a bad boy! (It was me. Added it for literally everything else except this. Don't know what I was doing!)